src/components/errorbars/compute_error.js

```diff
@@ -0,0 +1,82 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+
+/**
+ * Error bar computing function generator
+ *
+ * N.B. This function does not clean the dataPt entries, non-numeric
+ * entries result in undefined *error*
+ *
+ * @param {object} opts error bar attributes
+ *
+ * @return {function} :
+ *      @param {numeric} dataVal error magnitude in the negative direction
+ *      @param {number} index index of dataPt in its corresponding data array
+ *      @return {array}
+ *        - error[0] : error magnitude in the negative direction
+ *        - error[1] : " " " " positive "
+ */
+module.exports = function makeComputeError(opts) {
+    var type = opts.type,
+        symmetric = opts.symmetric;
+
+    if(type === 'data') {
+        var array = opts.array,
+            arrayminus = opts.arrayminus;
+
+        if(symmetric || arrayminus === undefined) {
+            return function computeError(dataPt, index) {
+                var val = +(array[index]);
+                return [val, val];
+            };
+        }
+        else {
+            return function computeError(dataPt, index) {
+                return [+arrayminus[index], +array[index]];
+            };
+        }
+    }
+    else {
+        var value = opts.value,
+            valueminus = opts.valueminus;
+
+        if(symmetric || valueminus === undefined) {
+            return function computeError(dataPt) {
+                var val = getErrorVal(type, dataPt, value);
```

Comment:
good point. all about the :racehorse: 

We have quite a few of these reusable calc functions, might as well set a good benchmark for them now.